extensions/arc/deployment/src/main/java/io/quarkus/arc/deployment/RuntimeBeanProcessor.java

```diff
@@ -82,6 +85,30 @@ public boolean test(BeanInfo bean) {
             producer.returnValue(result);
         }
         classCreator.close();
-        recorder.initSupplierBeans(map);
+        return new BeanInfoBuildItem(staticBeans, runtimeBeans);
+    }
+
+    @BuildStep
+    @Record(ExecutionTime.STATIC_INIT)
+    void initStatic(BeanInfoBuildItem infoBuildItem,
+            ArcRecorder recorder) {
+        recorder.initStaticSupplierBeans(infoBuildItem.staticBeans);
+    }
+
+    @BuildStep
+    @Record(ExecutionTime.RUNTIME_INIT)
+    void initRuntime(BeanInfoBuildItem infoBuildItem,
+            ArcRecorder recorder) {
+        recorder.initRuntimeSupplierBeans(infoBuildItem.runtimeBeans);
+    }
+
+    static final class BeanInfoBuildItem extends SimpleBuildItem {
```

Comment:
I'd like to rename this build item to something like `RuntimeBeanSuppliersBuildItem`. `BeanInfo` is confusing because of `io.quarkus.arc.processor.BeanInfo`.

Reply:
I think it would be better but I would drop the `Runtime` as we also have the static init ones and it could be confusing (I understand you don't mean runtime in this sense but that's a bit confusing IMHO).

In any case, it's local to this class so that's not a big issue.

Reply:
This build item is package-private so it's probably not a big problem. The `RuntimeBeanBuildItem` is de facto part of the API and the name is confusing. However, 1) we would break backward compatibility and 2) I'm not quite sure what the correct name should be...